crates/modelardb_server/src/metadata/mod.rs

```diff
@@ -1105,6 +1406,494 @@ mod tests {
+
+        let returned_files = create_metadata_manager_with_model_table_save_files_and_get_files(
+            &compressed_files,
+            start_time,
+            end_time,
+            min_value,
+            max_value,
+        )
+        .await;
+
+        compressed_files.len() == returned_files.len()
+    }
+
+    #[tokio::test]
+    async fn test_compressed_files_for_column_in_model_table_with_minimum_value() {
+        let (compressed_files, returned_files) =
+            create_metadata_manager_with_model_table_save_seven_files_and_get_files(
+                None,
+                None,
+                Some(80.0),
+                None,
+            )
+            .await;
+
+        assert_eq!(2, returned_files.len());
+        assert_eq!(compressed_files[3].name, returned_files[0]);
+        assert_eq!(compressed_files[5].name, returned_files[1]);
+    }
+
+    #[tokio::test]
+    async fn test_compressed_files_for_column_in_model_table_with_maximum_value() {
+        let (compressed_files, returned_files) =
+            create_metadata_manager_with_model_table_save_seven_files_and_get_files(
+                None,
+                None,
+                None,
+                Some(80.0),
+            )
+            .await;
+
+        assert_eq!(6, returned_files.len());
+        assert_eq!(compressed_files[0].name, returned_files[0]);
+        assert_eq!(compressed_files[1].name, returned_files[1]);
+        assert_eq!(compressed_files[2].name, returned_files[2]);
+        assert_eq!(compressed_files[4].name, returned_files[3]);
+        assert_eq!(compressed_files[5].name, returned_files[4]);
+        assert_eq!(compressed_files[6].name, returned_files[5]);
+    }
+
+    #[tokio::test]
+    async fn test_compressed_files_for_column_in_model_table_with_minimum_and_maximum_value() {
+        let (compressed_files, returned_files) =
+            create_metadata_manager_with_model_table_save_seven_files_and_get_files(
+                None,
+                None,
+                Some(75.0),
+                Some(80.0),
+            )
+            .await;
+
+        assert_eq!(1, returned_files.len());
+        assert_eq!(compressed_files[5].name, returned_files[0]);
+    }
+
+    async fn create_metadata_manager_with_model_table_save_seven_files_and_get_files(
+        start_time: Option<Timestamp>,
+        end_time: Option<Timestamp>,
+        min_value: Option<Value>,
+        max_value: Option<Value>,
+    ) -> (Vec<CompressedFile>, Vec<Uuid>) {
+        let compressed_files = SEVEN_COMPRESSED_FILES.to_vec();
+
+        let returned_files = create_metadata_manager_with_model_table_save_files_and_get_files(
+            &compressed_files,
+            start_time,
+            end_time,
+            min_value,
+            max_value,
+        )
+        .await;
+
+        (compressed_files, returned_files)
+    }
+
+    async fn create_metadata_manager_with_model_table_save_files_and_get_files(
```

Comment:
I do not think that this is a problem that can easily be fixed but I do think that the tests are harder to understand when we rely heavily on these very specific undocumented utility functions like this one.

Reply:
It might also make the tests harder to understand when we include these functions in the middle of tests. It might be easier to get an overview of what utility functions are used and whether some can be combined/refactored if they were defined together after the tests. 

Since we have used the convention of putting test utility functions in the middle of the tests elsewhere, this might be something that should be discussed before a decision is made.

Reply:
I agree and have added a task.